src/ScottPlot/Regression.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ScottPlot
+{
+	public interface IRegressionLine {
+		double[] GetCoefficients();
+		double[] Residual();
+		Plot Draw(Plot plot);
+	}
+
+	public class LinearRegressionLine : IRegressionLine {
+		private double[] x;
+		private double[] y;
+		private double[] coefficients;
+
+		public LinearRegressionLine(double[] xAxis, double[] yAxis) {
+			if (xAxis.Length != yAxis.Length) {
+				throw new ArgumentException("Axes of mismatched length");
+			}
+
+			x = xAxis;
+			y = yAxis;
+			coefficients = GetCoefficients();
+		}
+
+		private double SetMean(double[] set) {
+			double total = 0;
+
+			foreach (double curr in set) {
+				total += curr;
+			}
+
+			return total / set.Length;
+		}
+
+		private double y_hat(double x) {//Value predicted by regression model
+			//double[] coefficients = GetCoefficients();
+
+			return coefficients[0] + coefficients[1] * x;
+		}
+
+		public double[] GetCoefficients() {//y_hat = a+bx
+			double a;
+			double b;
+
+			double xMean=SetMean(x);
+			double yMean=SetMean(y);
+
+			double sumXYResidual = 0;
+			double sumXSquareResidual = 0;
+
+			for (int i = 0; i < x.Length; i++)
+			{
+				sumXYResidual += (x[i] - xMean) * (y[i] - yMean);
+				sumXSquareResidual += Math.Pow((x[i] - xMean), 2);
+			}
+
+			b = sumXYResidual / (sumXSquareResidual);
+
+			a = yMean - (b * xMean);//LSRL always passes through the point (x̅,y̅)
+
+			return new double[] { a, b };
+		}
+
+		public double[] Residual() {
+			//double[] coefficients = GetCoefficients();
+
+			double[] residuals = new double[y.Length];
+
+			for (int i = 0; i < y.Length; i++) {
+				residuals[i] = y[i] - y_hat(x[i]);
+			}
+
+			return residuals;
+		}
+
+		public Plot Draw(Plot plot) {
```

Comment:
Actually, I think `Regression` would make a good `Plottable`. It would make it easy to toggle its visibility on/off too. I can help to make the code changes to make this happen.